src/boot/efi/util.c

```diff
@@ -159,8 +162,13 @@ EFI_STATUS efivar_get_raw(const EFI_GUID *vendor, const CHAR16 *name, CHAR8 **bu
 
         err = uefi_call_wrapper(RT->GetVariable, 5, (CHAR16*) name, (EFI_GUID *)vendor, NULL, &l, buf);
         if (!EFI_ERROR(err)) {
-                *buffer = buf;
+
+                if (buffer)
+                        *buffer = buf;
+                else
+                        FreePool(buf);
```

Comment:
We actually have FreePoolp, so
```C
if (buffer)
      *buffer = TAKE_PTR(buf);
```
is possible.

Reply:
we have no TAKE_PTR right now in the EFI sources though. will add